components/tools/OmeroPy/src/omero/plugins/ldap.py

```diff
@@ -163,7 +166,15 @@ def getdn(self, args):
         if dn is not None and dn.strip():
             self.ctx.out(dn)
 
-    @admin_only
+    @admin_only(AdminPrivilegeModifyGroup)
+    def update_group(self, iupdate, group):
+        iupdate.saveObject(group)
+
+    @admin_only(AdminPrivilegeModifyUser)
+    def update_user(self, iupdate, user):
+        iupdate.saveObject(user)
+
+    @admin_only()
```

Comment:
In chatting with @mtbc, it turns out I had this backwards: I assume `@admin_only` would require a _full_ admin, but it doesn't. It simply defines "in system group" and therefore even a light admin with no privileges will be able to attempt the given call. Since `update_group` or `upgrade_user` then impose stricter requirements, the user will still get a sensible message.